packages/react-composites/tests/browser/chat/app/fake-back-end/FakeChatClient.ts

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { nanoid } from 'nanoid';
+import {
+  ChatParticipant,
+  ChatThreadClient,
+  ChatThreadItem,
+  CreateChatThreadRequest,
+  CreateChatThreadOptions,
+  CreateChatThreadResult
+} from '@azure/communication-chat';
+import { CommunicationIdentifier, getIdentifierKind } from '@azure/communication-common';
+import { PagedAsyncIterableIterator } from '@azure/core-paging';
+import { toFlatCommunicationIdentifier } from '@internal/acs-ui-common';
+import { Model } from './Model';
+import { IChatClient, IChatThreadClient, Thread } from './types';
+import { FakeChatThreadClient } from './FakeChatThreadClient';
+import { chatToSignalingParticipant, pagedAsyncIterator, latestMessageTimestamp } from './utils';
+import { BaseChatThreadEvent } from '@azure/communication-signaling';
+import { getThreadEventTargets } from './ThreadEventEmitter';
+
+/**
+ * A public interface compatible stub for ChatClient.
+ */
+export class FakeChatClient implements IChatClient {
+  private realtimeNotificationsEnabled = false;
+  private threadClients: FakeChatThreadClient[] = [];
+
+  constructor(private model: Model, private userId: CommunicationIdentifier) {}
+
+  getChatThreadClient(threadId: string): ChatThreadClient {
+    this.model.checkedGetThread(this.userId, threadId);
+    const threadClient = new FakeChatThreadClient(this.model, this.userId, threadId);
+    this.threadClients.push(threadClient);
+    return threadClient as IChatThreadClient as ChatThreadClient;
+  }
+
+  createChatThread(
+    request: CreateChatThreadRequest,
+    options?: CreateChatThreadOptions
+  ): Promise<CreateChatThreadResult> {
+    const now = new Date(Date.now());
+    const participants = this.withCurrentUserInThread(options?.participants ?? []);
+    const thread: Thread = {
+      id: nanoid(),
+      version: 0,
+      createdOn: now,
+      createdBy: getIdentifierKind(this.userId),
+      topic: request.topic,
+      participants,
+      messages: [
+        {
+          id: '0',
+          type: 'participantAdded',
+          sequenceId: '0',
+          version: '0',
+          createdOn: new Date(),
```

Comment:
should we use the now variable here or do we want a slightly later time?

Reply:
For now, it is fine because there are no messages once the FakeClient is created so it is guaranteed to be the first message in the thread.